src/ha-utils/ha/gcp/services/LoadBalancerService.go

```diff
@@ -0,0 +1,120 @@
+package services
+
+import (
+	"encoding/json"
+	"ha-utils/ha/common/models"
+	"ha-utils/ha/common/constants"
+	commoninterfaces "ha-utils/ha/common/interfaces"
+	gcpmodels "ha-utils/ha/gcp/models"
+	gcputils "ha-utils/ha/gcp/utils"
+	"log"
+	"strings"
+)
+
+type LoadBalancerService struct {
+	svc commoninterfaces.IServiceClient
+}
+
+func (lbService *LoadBalancerService) Initialize(params ...interface{}) {
+
+	lbService.svc = params[0].(commoninterfaces.IServiceClient)
+
+}
+
+func (lbService *LoadBalancerService) GetLoadBalancer(lbName string, regionName string) (*gcpmodels.LoadBalancer, bool) {
+
+	var loadBalancerAPIUrl, responseStr, responseCode string
+	var returnValue bool
+	var loadBalancer *gcpmodels.LoadBalancer = &gcpmodels.LoadBalancer{}
+	var params models.IaaSDescriptors = lbService.svc.GetIaaSDescriptors()
+
+	loadBalancerAPIUrl = params.ManagementURL + "/compute/v1/projects/" + params.ProjectId + "/regions/" + regionName + "/backendServices/" + lbName
+	/*
+		The REST API URL format is as follows.
+		// https://www.googleapis.com/compute/v1/projects/sap-picpcore-gcp-dev8/regions/europe-west1/backendServices/postgres-ha-test-lb1
+		GET https://www.googleapis.com/compute/v1/projects/{project}/global/backendServices/{resourceId}
+	*/
+	responseStr, responseCode, returnValue = lbService.svc.InvokeAPI("GET", loadBalancerAPIUrl, lbService.svc.GetCommonRequestHeaders(), nil)
+	if returnValue == true {
+		err := json.Unmarshal([]byte(responseStr), loadBalancer)
+		if err != nil {
+			log.Println("Error occurred while unmarshalling load balancer details ", err.Error())
+			return nil, false
+		}
+		log.Println("Load Balancer with name :", lbName, "and link :", loadBalancer.SelfLink, "retrieved successfully.")
+		return loadBalancer, true
+	} else {
+		if strings.Compare(responseCode, constants.HTTP_STATUS_NOT_FOUND) != 0 {
+			// if GetLoadBalancer() is called before the given health check is created, 404 status will be returned.
+			// If the error is something other than 404, then something wrong might've happened. Lets log it explicitly for later analysis.
+			log.Println("Error occurred during load balancer get call : HTTP Status: ", responseCode, " error: ", responseStr)
+			return nil, false
+		} else {
+			// 404. It could be the case that a load balancer is yet to be created. Lets return true.
```

Comment:
need more explaination here. What if LB is truly not there? Even in that case would 404 be treated as true?

Reply:
Yes. It will be deemed as true. If the HA feature is enabled, this piece of code is triggered, and if LoadBalancer is still not present, then we always create a LB. 
Should something be changed here?